### DISCUSSION TITLE
New update has removed all of my maps

### DISCUSSION THREAD

**Discussion by Tremayah** [ppy/osu#24816] [Q&A] [answered]
About a year ago I switched to my laptop to play osu for a while, and when I came back to my pc all of my beatmaps and scores were gone, and since then I have been slowly building up the collection again. Along with the beatmaps the scores, skins, and collections were also gone.

Yesterday I played osu and everything was as normal, with my newer collection of beatmaps that I had been collecting over the past while. Today there is an update, and all of my beatmaps from the last year are gone, and the ones from a year ago are all back, along with the collections. Now this sucks because I was just getting back on my feet and found some maps I really enjoy and now they're gone.

I reload the game to see if that helps, and it makes things considerably worse. I now have 0 beatmaps. I can see them all in the files folder, but apparently lazer can't read that...

Is there a way to get my most recent collection of maps back, from before the update that happened in the last 24 hours or so?
[network.log](https://github.com/ppy/osu/files/12613822/network.log)
[performance.log](https://github.com/ppy/osu/files/12613823/performance.log)
[runtime.log](https://github.com/ppy/osu/files/12613824/runtime.log)
[updater.log](https://github.com/ppy/osu/files/12613825/updater.log)
[database.log](https://github.com/ppy/osu/files/12613826/database.log)
[input.log](https://github.com/ppy/osu/files/12613827/input.log)
[legacy-ipc.log](https://github.com/ppy/osu/files/12613828/legacy-ipc.log)

**Comment by peppy** [comment]
can you please take a screenshot of your osu folder? there should be some client.realm files in there that you can potentially recover from

**Reply by Tremayah** [reply]
![image](https://github.com/ppy/osu/assets/125309136/747d7e48-a8c0-4c4d-9e68-030dccc1a0bd)
![image](https://github.com/ppy/osu/assets/125309136/c89cb063-39a5-4005-a64e-4abdcf759376)
![image](https://github.com/ppy/osu/assets/125309136/58347bc2-d6ac-4e09-87c2-467ca519b0d3)

**Reply by rodrigopina360** [reply]
Not sure how the realm system works, but that client_newer_version.realm might be a backup just before you updated, backup the whole folder somewhere (please do 2-3 backups just to be sure) and then try deleting client.realm, and renaming that client_newer_version.realm to client.realm. With luck the maps might come back. If not I guess it's waiting for ppy's response

**Reply by Tremayah** [reply]
Nope didn't work :( ty for trying

Will await PPY's response

**Reply by peppy** [reply]
If the above didn't work, you'll have to upload the files for me to take a look. Good thing is you have plenty of backups created by osu! for you.

**Reply by Tremayah** [reply]
Sorry please could you be more specific with what files you want uploaded? The majority of file types in appdata>roaming>osu aren't supported by github such as REALM, LOCK, and BACKUP-LOG. I'm not very savvy with this stuff so please do be specific if possible.

**Reply by rodrigopina360** [reply]
He probably wants the .realm files (if you zip them you should be able to send them)

**Reply by Tremayah** [reply]
[Realm Files For Osu.zip](https://github.com/ppy/osu/files/12643508/Realm.Files.For.Osu.zip)

**Reply by peppy** [reply]
I've tested locally and the latest backup you provided works fine.

Here's the steps:

- Delete ALL `client.*` files from your osu folder.
- **Don't delete the `files` directory**
- Download the attached `client.realm` and put it in your osu folder (i've upgraded it for you to the latest version)

[client.realm.zip](https://github.com/ppy/osu/files/12645226/client.realm.zip)

**Reply by Tremayah** [reply]
Here is what I deleted:
![image](https://github.com/ppy/osu/assets/125309136/e4a520ca-aede-4130-8269-6c75476d7775)

And here is what my osu folder now looks like:
![image](https://github.com/ppy/osu/assets/125309136/b1a5c0b9-fdce-4b60-b6fe-bcf82392182d)

I still have no maps. Do I need to delete the client.realm.management folder aswell? idk if "files" usually includes folders...

**Reply by peppy** [reply]
You didn't extract the zip file I sent you.. You need to extract it.

**Reply by Tremayah** [reply]
I extracted the zip file into the osu folder, should I have extracted it into osu > client.realm? After extracting the zipped file still remained but the 2 client.realm files are there again, so it definitely did extract.

**Reply by peppy** [reply]
sorry but it all works locally so i can only assume you've done some step incorrectly. are you able to have someone else try this on your PC?

**Reply by rodrigopina360** [reply]
Not sure if helpful, but I can try to replace my files (after a backup) to see if I can make them work

**Reply by peppy** [reply]
I'll try one last time:

- Delete EVERY client or realm related file from the folder.
- Download the archive I attached above.
- Extract it into the osu! folder while osu! isn't running
- You should only have a single file with `client` in the filename, and it should be `client.realm`, around 8mb.

**Comment by Tremayah** [comment]
I deleted anything that had the words client or realm that I could see in the osu folder
I then downloaded the file you attached and extracted it to the osu folder
![image](https://github.com/ppy/osu/assets/125309136/2e2b00ad-4471-44d3-81af-71318c7367a4)

I launch Osu and this error pops up
![image](https://github.com/ppy/osu/assets/125309136/321996b7-aa8e-498f-befb-90ef00f0c849)

None of my maps are present when I go into singleplayer 

Should I backup the files folder and then just re install Osu?

**Reply by Tremayah** [reply]
![image](https://github.com/ppy/osu/assets/125309136/0ae97f73-f9ce-49ef-bb20-d3e43e21f764)
This is what my osu folder looks like after extraction and running Osu

**Reply by peppy** [reply]
This means your osu! install is an old version.

Just run the installer, then follow the same process above again. **Do NOT UNINSTALL**.

https://github.com/ppy/osu/releases/download/2023.914.0/install.exe

**Reply by Tremayah** [reply]
Awesome I have the 1100 maps from over a year ago that disappeared a while back, but the maps I had recently, eg before the Lazer 2023.914 update are still gone. I'm happy I've got a portion of my maps back, is there a way to get the recent maps and scores back? Thanks for the help so far <3

**Reply by peppy** [reply, marked as answer]
You'll have to try the other backup files one by one to figure which has the version you want.

You can rename any of the files you provided to me to `client.realm` to test them (if you already deleted them locally you still have the archive you posted earlier in the thread).

Let's call this resolved, though.

**Reply by rszyma** [reply]
I've built osu! from source with Release flag and ran it just to test something. Now when I normally run the latest official release osu! version it don't show any beatmaps. I suppose the `client.realm` stuff got updated when I updated the osu version. @peppy could please you give a general instructions how to recover the beatmaps to previous (official) version?

Edit:
If perhaps this will get fixed automatically once the new official build is released it's okay, I can wait.

Edit2: can confirm - waiting for a new release fixed the database

**Reply by peppy** [reply]
yes just wait i guess:
